mongo/client_examples_test.go

```diff
@@ -0,0 +1,200 @@
+// Copyright (C) MongoDB, Inc. 2017-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package mongo_test
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/mongo/readpref"
+)
+
+func ExampleConnect_ping() {
+	// Create a Client to a MongoDB server and use Ping to verify that the server is running.
+
+	clientOpts := options.Client().ApplyURI("mongodb://localhost:27017")
+	client, err := mongo.Connect(context.TODO(), clientOpts)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer func() {
+		if err = client.Disconnect(context.TODO()); err != nil {
+			log.Fatal(err)
+		}
+	}()
+
+	// Call Ping to verify that the deployment is up and the Client was configured successfully.
+	// As mentioned in the Ping documentation, this reduces application resiliency as the server may be
+	// temporarily unavailable when Ping is called.
+	if err = client.Ping(context.TODO(), readpref.Primary()); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func ExampleConnect_replicaSet() {
+	// Create and connect a Client to a replica set deployment.
+	// Given this URI, the Go driver will first communicate with localhost:27017 and use the response to discover
+	// any other nodes in the replica set.
+
+	clientOpts := options.Client().ApplyURI("mongodb://localhost:27017/?replicaSet=replset")
+	client, err := mongo.Connect(context.TODO(), clientOpts)
+	if err != nil {
+		log.Fatal(err)
+	}
+	_ = client
+}
+
+func ExampleConnect_sharded() {
+	// Create and connect a Client to a sharded deployment.
+	// The URI for a sharded cluster should specify all mongos nodes in the cluster.
+
+	clientOpts := options.Client().ApplyURI("mongodb://localhost:27017,localhost:27018")
+	client, err := mongo.Connect(context.TODO(), clientOpts)
+	if err != nil {
+		log.Fatal(err)
+	}
+	_ = client
+}
+
+func ExampleConnect_sRV() {
+	// Create and connect a Client using an SRV record.
+	// SRV records allow administrators to configure a single domain to return a list of host names.
+	// The driver will resolve SRV records prefixed with "_mongodb_tcp" and use the returned host names to
+	// build its view of the deployment.
+	// See https://docs.mongodb.com/manual/reference/connection-string/ for more information about SRV.
+
+	// Versions 1.1.0 or later of the driver support SRV polling for mongos discovery. Prior to this version, using
```

Comment:
This seems more like specification that example code.  Just double-checking that everyone is comfortable with it being here rather than directly in the GoDoc for types/functions.

CC @xdg

Reply:
I talked to @xdg in person and we decided that the current wording is too much, but some sort of note would be good to have as the docs page for connection string format doesn't talk about SRV polling. I changed it to a much shorter note ("Full support for SRV records with sharded clusters requires driver version 1.1.0 or higher").